File: Subprogramas/Aninhados/Aninhados.pas
//Pascal
program Aninhados;

    procedure ProcedimentoExterno;
        procedure ProcedimentoInterno;
        begin
            WriteLn('Este é o procedimento interno.');
        end;

    begin
        WriteLn('Este é o procedimento externo.');
        ProcedimentoInterno;  // Chama o procedimento interno
    end;

begin
    ProcedimentoExterno;  // Chama o procedimento externo
end.

File: Conceitos de LPs/Subprogramas/Aninhados/Aninhados.cs
//C Sharp (C#)
using System;

class Aninhados
{
    static void MetodoExterno()
    {
        Console.WriteLine("Este é o método externo.");

        void MetodoInterno()
        {
            Console.WriteLine("Este é o método interno.");
        }

        MetodoInterno();  // Chama o método interno
    }

    static void Main()
    {
        MetodoExterno();  // Chama o método externo
    }
}
